scipy/special/cython_special.pyx

```diff
@@ -3175,13 +3167,29 @@ cpdef double nbdtrin(double x0, double x1, double x2) noexcept nogil:
     """See the documentation for scipy.special.nbdtrin"""
     return _func_nbdtrin(x0, x1, x2)
 
-cpdef double ncfdtr(double x0, double x1, double x2, double x3) noexcept nogil:
+cpdef df_number_t ncfdtr(df_number_t x0, df_number_t x1, df_number_t x2, df_number_t x3) noexcept nogil:
     """See the documentation for scipy.special.ncfdtr"""
-    return _func_ncfdtr(x0, x1, x2, x3)
+    if df_number_t is float:
+        return (<float(*)(float, float, float, float) noexcept nogil>scipy.special._ufuncs_cxx._export_ncf_cdf_float)(x0, x1, x2, x3)
+    elif df_number_t is double:
+        return (<double(*)(double, double, double, double) noexcept nogil>scipy.special._ufuncs_cxx._export_ncf_cdf_double)(x0, x1, x2, x3)
+    else:
+        if df_number_t is double:
+            return NAN
+        else:
+            return NAN
```

Comment:
```suggestion
        return NAN
```
This is copied from elsewhere, but it looks really redundant. If the rest looks good, I'll merge as-is, but consider consolidating or adding a comment.